lib/TelemetryFeed.jsm

```diff
@@ -577,32 +577,28 @@ this.TelemetryFeed = class TelemetryFeed {
    * 3). In shield experiments conducted in release, it collects client_id and message_id
    */
   applyCFRPolicy(ping) {
+    let excludeClientID;
```

Comment:
I'm not sure about these changes that add `excludeClientID`, the logic is simple but it's difficult to follow through the code.
The behavior of `apply*Policy` now changed, but the name hasn't and it's only half applying the policy and half deferring for later in `_createStructuredIngestionPing`.
Could we change things around so that in `handleASRouterUserEvent` we don't immediately call `createASRouterEvent` and instead do it later in `_createStructuredIngestionPing`?
We could instead of passing `excludeClientID`, pass the policy name that should be applied (or the `event.action`).

Reply:
Thanks for pointing it out! Agreed, this `excludeClientID` thing complicated the code substantially. I've refactored it out so that the logic of "shall we use `client_id` or `impression_id`?" was completely decided by those `applyXYZPolicy` functions.